### PR TITLE
test: add typescript test

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Adding a test to make sure the module works with TypeScript as expected.  Covers #177 and #178 and confuses me with #179.

### CODE REVIEW

File: test/fixtures/kitchen/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "clean": "gts clean",
+    "compile": "tsc -p .",
+    "fix": "gts fix",
+    "prepare": "npm run compile",
+    "pretest": "npm run compile",
+    "posttest": "npm run check",
+    "start": "node build/src/index.js"
+  },
+  "keywords": [],
+  "author": "Justin Beckwith <justin.beckwith@gmail.com>",
```
Comment (ofrobots) [inline test/fixtures/kitchen/package.json]: Google

File: test/fixtures/kitchen/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "license": "MIT",
```
Comment (ofrobots) [inline test/fixtures/kitchen/package.json]: Use Apache 2.0, same as the parent project.

File: test/fixtures/kitchen/package.json (hunk 1 above)
Comment (ofrobots) [reply test/fixtures/kitchen/package.json]: better yet, remove these fields.

Comment (JustinBeckwith) [reply test/fixtures/kitchen/package.json]: done

File: test/fixtures/kitchen/package.json (hunk 2 above)
Comment (JustinBeckwith) [reply test/fixtures/kitchen/package.json]: done

File: test/test.kitchen.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const spawnp = (command: string, args: string[], options: cp.SpawnOptions = {}):
+    Promise<void> => {
+      return new Promise((resolve, reject) => {
+        cp.spawn(command, args, Object.assign(options, {stdio: 'inherit'}))
+            .on('close',
+                (code, signal) => {
+                  if (code === 0) {
+                    resolve();
+                  } else {
+                    reject(`Spawn failed with an exit code of ${code}`);
```
Comment (ofrobots) [inline test/test.kitchen.ts]: nit: Strings are not Errors.

**Pull request merged**